backend/controller/observability/fsm.go

```diff
@@ -2,53 +2,91 @@ package observability
 
 import (
 	"context"
+	"errors"
 	"fmt"
 
 	"go.opentelemetry.io/otel"
 	"go.opentelemetry.io/otel/attribute"
 	"go.opentelemetry.io/otel/metric"
+	"go.opentelemetry.io/otel/metric/noop"
 
 	"github.com/TBD54566975/ftl/backend/schema"
 	"github.com/TBD54566975/ftl/internal/observability"
 )
 
 const (
-	fsmMeterName    = "ftl.fsm"
-	fsmRefAttribute = "ftl.fsm.ref"
+	fsmMeterName             = "ftl.fsm"
+	fsmRefAttribute          = "ftl.fsm.ref"
+	fsmDestStateRefAttribute = "ftl.fsm.dest.state.ref"
 )
 
-var fsmMeter = otel.Meter("ftl.fsm")
+type FSMMetrics struct {
+	meter              metric.Meter
+	instancesActive    metric.Int64UpDownCounter
+	transitionsActive  metric.Int64UpDownCounter
+	transitionAttempts metric.Int64Counter
+}
 
-var fsmCounters = struct {
-	instancesActive metric.Int64UpDownCounter
-}{}
+func initFSMMetrics() (*FSMMetrics, error) {
+	result := &FSMMetrics{}
 
-func InitFSMMetrics() error {
+	var errs error
 	var err error
 
-	fsmCounters.instancesActive, err = fsmMeter.Int64UpDownCounter(
-		fmt.Sprintf("%s.instances.active", fsmMeterName),
-		metric.WithDescription("counts the number of active FSM instances"))
+	result.meter = otel.Meter("ftl.fsm")
 
-	if err != nil {
-		return fmt.Errorf("could not initialize fsm metrics: %w", err)
+	counter := fmt.Sprintf("%s.instances.active", fsmMeterName)
+	if result.instancesActive, err = result.meter.Int64UpDownCounter(
+		counter,
+		metric.WithDescription("counts the number of active FSM instances")); err != nil {
+		errs = errors.Join(errs, fmt.Errorf("%q counter init failed; falling back to noop: %w", counter, err))
```

Comment:
Just to dedup the repeated code, what do you think about something like:
```
    errs = joinInitErrors(errs, err, counter)
    ...
}

func joinInitErrors(...) error {
    return errors.Join(errs, fmt.Errorf("%q counter init failed; falling back to noop: %w", counter, err))
}
```

Reply:
adopted